libc/utils/MPFRWrapper/MPFRUtils.cpp

```diff
@@ -439,7 +441,10 @@ class MPFRNumber {
 
   MPFRNumber sinpi() const {
     MPFRNumber result(*this);
-    mpfr_sinpi(result.value, value, mpfr_rounding);
+    MPFRNumber value_pi(0.0, 1280);
```

Comment:
you can use predefined macros `MPFR_VERSION_MAJOR` and `MPFR_VERSION_MINOR` to see if `mpfr_sinpi` could be used to speed up the computations